tools/src/main/scala/scala/scalanative/build/NativeConfig.scala

```diff
@@ -347,6 +365,14 @@ object NativeConfig {
       copy(embedResources = value)
     }
 
+    def withResourceIncludePatterns(value: Seq[String]): NativeConfig = {
+      copy(resourceIncludePatterns = value)
+    }
+
+    def withResourceExcludePatterns(value: Seq[String]): NativeConfig = {
+      copy(resourceExcludePatterns = value)
+    }
+
```

Comment:
We can add variants for appending a single pattern, these might be sometimes easier to use, eg: 
```Scala
 config
  .withResourceIncludePattern("foo")
  .withResourceIncludePatterns(config.resourceIncludePatterns :+ "foo")
```
However, at that time we would need to handle the default `**` pattern. We can do it here by checking if `resourceIncludePatterns eq DefaultIncludePattern` and then replace the sequence, or otherwise append to it.

Reply:
I personally feel having two APIs: "appending a single pattern" and "replace the patterns with the given seq" seems confusing, and even complicated with the special handling with replacing the default pattern.
Maybe we can compromise with the API that takes varargs instead of Seq so users can write something like `.withResourceIncludePatterns("foo")`?

Reply:
Let's stay with the current solution then using Seq, that way it would be aligned with remaining settings